--- mozc/src/gyp/common.gypi
+++ mozc/src/gyp/common.gypi
@@ -105,7 +105,6 @@
     ],
     # Libraries for GNU/Linux environment.
     'linux_ldflags': [
-      '-lc++',
       '-pthread',
     ],
 
@@ -213,7 +212,6 @@
           ['compiler_target=="clang" or compiler_target=="gcc"', {
             'cflags_cc': [
               '-std=c++17',
-              '-stdlib=libc++',
             ],
           }],
         ],
@@ -233,7 +231,6 @@
           ['compiler_host=="clang" or compiler_host=="gcc"', {
             'cflags_cc': [
               '-std=c++17',
-              '-stdlib=libc++',
             ],
           }],
         ],
@@ -409,12 +406,12 @@
   'conditions': [
     ['target_platform=="Linux"', {
       'make_global_settings': [
-        ['AR', '<!(which ar)'],
-        ['CC', '<!(which clang)'],
-        ['CXX', '<!(which clang++)'],
-        ['LD', '<!(which ld)'],
-        ['NM', '<!(which nm)'],
-        ['READELF', '<!(which readelf)'],
+        ['AR', '<!(which ${AR_target:-ar})'],
+        ['CC', '<!(which ${CC_target:-cc})'],
+        ['CXX', '<!(which ${CXX_target:-c++})'],
+        ['LD', '<!(which ${LD_target:-ld})'],
+        ['NM', '<!(which ${NM_target:-nm})'],
+        ['READELF', '<!(which ${READELF_target:-readelf})'],
       ],
     }],
     ['target_platform=="Android"', {
@@ -459,12 +456,12 @@
         ['LD', '<(ndk_bin_dir)/<(toolchain_prefix)-ld'],
         ['NM', '<(ndk_bin_dir)/<(toolchain_prefix)-nm'],
         ['READELF', '<(ndk_bin_dir)/<(toolchain_prefix)-readelf'],
-        ['AR.host', '<!(which ar)'],
-        ['CC.host', '<!(which clang)'],
-        ['CXX.host', '<!(which clang++)'],
-        ['LD.host', '<!(which ld)'],
-        ['NM.host', '<!(which nm)'],
-        ['READELF.host', '<!(which readelf)'],
+        ['AR.host', '<!(which ${AR:-ar})'],
+        ['CC.host', '<!(which ${CC:-cc})'],
+        ['CXX.host', '<!(which ${CXX:-c++})'],
+        ['LD.host', '<!(which ${LD:-ld})'],
+        ['NM.host', '<!(which ${NM:-nm})'],
+        ['READELF.host', '<!(which ${READELF:-readelf})'],
       ],
     }],
   ],
