--- client/X11/xf_keyboard.c
+++ client/X11/xf_keyboard.c
@@ -154,6 +154,11 @@ void xf_keyboard_key_press(xfContext* xfc, BYTE keycode, KeySym keysym)
 	if (keycode < 8)
 		return;
 
+	if (keycode == 9)
+		keycode = 66;
+	else if (keycode == 66)
+		keycode = 9;
+
 	xfc->KeyboardState[keycode] = TRUE;
 
 	if (xf_keyboard_handle_special_keys(xfc, keysym))
@@ -167,6 +172,11 @@ void xf_keyboard_key_release(xfContext* xfc, BYTE keycode)
 	if (keycode < 8)
 		return;
 
+	if (keycode == 9)
+		keycode = 66;
+	else if (keycode == 66)
+		keycode = 9;
+
 	xfc->KeyboardState[keycode] = FALSE;
 
 	xf_keyboard_send_key(xfc, FALSE, keycode);
